openlibrary/plugins/importapi/import_validator.py

```diff
@@ -1,24 +1,185 @@
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import Annotated, Any, Optional, TypeVar
 
 from annotated_types import MinLen
-from pydantic import BaseModel, ValidationError
+from pydantic import (
+    AnyHttpUrl,
+    BaseModel,
+    ConfigDict,
+    Field,
+    PositiveInt,
+    RootModel,
+    StringConstraints,
+    ValidationError,
+)
 
 T = TypeVar("T")
 
 NonEmptyList = Annotated[list[T], MinLen(1)]
-NonEmptyStr = Annotated[str, MinLen(1)]
+NonEmptyStr = Annotated[str, StringConstraints(strip_whitespace=True, min_length=1)]
+
+
+class PublishCountry(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^[a-z]{2,3}$')] = Field(
+        ...,
+        description='The MARC21 country code. See https://www.loc.gov/marc/countries/cou_home.html',
+        examples=['enk', 'gw', 'flu'],
+    )
+
+
+class LanguageCode(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^[a-z]{3}$')] = Field(
+        ...,
+        description='The MARC21 language code. See https://www.loc.gov/marc/languages/language_code.html',
+        examples=['eng', 'fre', 'ger', 'tha'],
+    )
+
+
+class Isbn10(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^([0-9][- ]*){9}[0-9X]$')]
+
+
+class Isbn13(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^([0-9][- ]*){13}$')]
+
+
+class WorkKey(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^/works/OL[0-9]+W$')]
+
+
+class Language(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+    key: Annotated[str, StringConstraints(pattern=r'^/languages/[a-z]{3}$')] = Field(
+        ..., examples=['/languages/eng', '/languages/ger']
+    )
+
+
+class LcClassification(RootModel):
+    root: NonEmptyStr = Field(
+        ...,
+        description='The Library of Congress Classification number. See https://www.loc.gov/catdir/cpso/lcc.html We include the imprint date as the last four digits.',  # noqa: E501
+        examples=['BS571.5 .S68 1995', 'Z673.D62 C65 1994'],
+    )
+
+
+class Key(Enum):
+    field_type_link = '/type/link'
+
+
+class Type(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+    key: Key
+
+
+class Link(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+    url: AnyHttpUrl
+    title: NonEmptyStr
+    type: Optional[Type] = None
+
+
+class EntityType(Enum):
+    person = 'person'
+    org = 'org'
+    event = 'event'
 
 
 class Author(BaseModel):
-    name: NonEmptyStr
+    model_config = ConfigDict(extra="forbid")
+
+    name: NonEmptyStr = Field(examples=['Hubbard, Freeman H.', 'Joan Miró'])
+    personal_name: Optional[NonEmptyStr] = Field(
+        None,
+        description="Can be identical to 'name'. TODO: provide information on the intended difference.",
+    )
+    birth_date: Optional[NonEmptyStr] = Field(None, examples=[])
+    death_date: Optional[NonEmptyStr] = Field(None, examples=[])
+    entity_type: Optional[EntityType] = None
```

Comment:
`entity_type` could use some examples:

https://github.com/internetarchive/openlibrary-client/blob/bf55435a92567a0341008a3c0eeb2637b7988c27/olclient/schemata/author.schema.json#L37-L40

Reply:
Was thinking of make that type more strict, making it only accept person

Reply:
Done!